vmupdate/qube_connection.py

```diff
@@ -248,21 +253,24 @@ def _collect_stderr(self, proc) -> bytes:
                         StatusInfo.updating(self.qube, progress))
                 else:
                     stderr += untrusted_line + b'\n'
-            else:
+            elif proc.poll() is not None:
```

Comment:
I can no longer reproduce the hang, I guess this fixed it :)

Reply:
I _guess_ the issue was stopping at the first empty line. If that's the case, it might be better to change the sentinel to `None` and this condition to `untrusted_line is None`. This way, you won't risk loosing last few lines of output printed just before process exit.

Reply:
I think both versions are wrong... What should really be here, is `b''` (not `''`) as the sentinel for the `iter()` call. I'll add a commit after a bit more testing.